pkg/docker/config/config_linux.go

```diff
@@ -0,0 +1,81 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/containers/image/pkg/keyctl"
+)
+
+func getAuthFromKernelKeyring(registry string) (string, string, error) {
+	userkeyring, err := keyctl.UserKeyring()
+
+	if err != nil {
+		return "", "", err
+	}
+	key, err := userkeyring.Search(genDescription(registry))
+	if err != nil {
+		return "", "", err
+	}
+	authData, err := key.Get()
+	if err != nil {
+		return "", "", err
+	}
+	parts := strings.SplitN(string(authData), ":", 2)
+	if len(parts) != 2 {
+		return "", "", nil
+	}
+	return parts[0], parts[1], nil
+}
+
+func deleteAuthFromKernelKeyring(registry string) error {
+	userkeyring, err := keyctl.UserKeyring()
+
+	if err != nil {
+		return err
+	}
+	key, err := userkeyring.Search(genDescription(registry))
+	if err != nil {
+		return err
+	}
+	return key.Unlink()
+}
+
+func setAuthToKernelKeyring(registry, username, password string) error {
+	keyring, err := keyctl.SessionKeyring()
+
+	if err == nil {
+		id, err := keyring.Add(genDescription(registry), []byte(fmt.Sprintf("%s:%s", username, password)))
+		if err != nil {
+			return err
+		}
+
+		// Set the user permisson to link the key to userkeyring later
+		err = keyctl.SetPerm(id, keyctl.PermUserAll)
```

Comment:
I think this call is worth a comment. `UserAll` can trick a reader into believing that all users are affected but it means that the current user has all rights for this specific key (view, read, write, search, link, set attribute).